float32 gain
---
int32 error